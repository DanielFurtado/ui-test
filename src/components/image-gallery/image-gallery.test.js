import ImageGallery from './image-gallery.component';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom';

const data = [
  {
  'id': 'abc123',
  'description': 'some description.',
  'alt_description': 'alternative description',
  'likes': 10,
  'user': {
    'name': 'Fred Mercury',
    'profile_image': 'https://www.some-link/images/profile-image-1'
  },
  'url': 'https://www.some-link/images/some-image-1'
  },
  {
    'id': 'def456',
    'description': 'some description 2.',
    'alt_description': 'alternative description 2',
    'likes': 20,
    'user': {
      'name': 'David Bowie',
      'profile_image': 'https://www.some-link/images/profile-image-2'
    },
    'url': 'https://www.some-link/images/url-image-2'
  },
];

describe('Image Gallery', ()=> {
  it('renders', () => {
    const component = renderer.create(<ImageGallery images={data} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders Image Gallery', () => {
    const {getByText} = render(<ImageGallery images={data} />);
    expect(getByText('Photo gallery')).toBeInTheDocument();
  });
});
