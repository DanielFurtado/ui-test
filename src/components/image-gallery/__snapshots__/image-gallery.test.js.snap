// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Gallery renders 1`] = `
<div
  className="image-gallery-container"
>
  <div
    className="image-gallery-heading"
  >
    <h1>
      Photo gallery
    </h1>
    <p>
      Some of our clients have shared some of their amazing photographic work with us here at Motorway! This is truly inspirational!
    </p>
  </div>
  <div
    className="image-gallery-wrapper"
  >
    <div
      className="card-container"
      data-testid="card-container"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(https://www.some-link/images/some-image-1.jpg)",
        }
      }
    >
      <div
        className="card-body-container"
      >
        <div
          className="card-info shine"
        >
          <p>
            Fred Mercury
          </p>
          <span
            className="arrow-right"
          />
        </div>
      </div>
    </div>
    <div
      className="card-container"
      data-testid="card-container"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(https://www.some-link/images/url-image-2.jpg)",
        }
      }
    >
      <div
        className="card-body-container"
      >
        <div
          className="card-info shine"
        >
          <p>
            David Bowie
          </p>
          <span
            className="arrow-right"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
